### PR TITLE
Publish image to Public ECR

### PR DESCRIPTION
**Pull request by zhonghui12**
*Issue #123 , if available:*

*Description of changes:*
Publish image onto ECR Public

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: scripts/publish.sh
Comment (PettitWesley) [inline scripts/publish.sh:123]: I don't think this is needed, the repo should already exist

Comment (zhonghui12) [reply scripts/publish.sh:123]: Yeah.. Make sense. Will remove this line.

File: scripts/publish.sh
Comment (hossain-rayhan) [inline scripts/publish.sh:49]: Did we create a new account?

Comment (zhonghui12) [reply scripts/publish.sh:49]: No, this is my account.. for testing purpose. Will modify this part after the test.

File: scripts/publish.sh
Comment (hossain-rayhan) [inline scripts/publish.sh:122]: I am not sure why its default to `us-east-1`? Is it for public ECE?

Comment (zhonghui12) [reply scripts/publish.sh:122]: Yes, for public ECR, we always set the region to `us-east-1`

File: scripts/publish.sh
Comment (hossain-rayhan) [reply scripts/publish.sh:123]: Ahh kind of confused? Can't we test before merging the code into mainline?

File: scripts/publish.sh
Comment (PettitWesley) [inline scripts/publish.sh:122]: for testing, use a different public repo. Create one in your own account and make the code publish to it.

File: scripts/publish.sh
Comment (hossain-rayhan) [reply scripts/publish.sh:49]: Ahh kind of confused. Can't we test before merging the code into mainline?

Comment (zhonghui12) [reply scripts/publish.sh:49]: Yes, we can. But for this PR, I didn't mean to merge it now. This is not the final version for it and this PR is just a way for members to leave comments. Sorry for the confusion..

File: scripts/publish.sh
Comment (hossain-rayhan) [reply scripts/publish.sh:122]: Maybe I don't have enough insights here. For public ECR is it necessary to hit a regional endpoint for getting login-password?

Comment (zhonghui12) [reply scripts/publish.sh:122]: Yes we should: https://docs.aws.amazon.com/AmazonECR/latest/public/getting-started-cli.html. We still need to authenticate to the public registry but specify `--region us-east-1`.

File: scripts/publish.sh
Comment (hossain-rayhan) [inline scripts/publish.sh:131]: Do we need this again? We got the login credentials in line 122.

Comment (zhonghui12) [reply scripts/publish.sh:131]: yes we need it. For some reason, we need to got it every time before we want to `create manifest list`. If we don't add this line, we would have an error.

Comment (hossain-rayhan) [reply scripts/publish.sh:131]: nit: Hmm, interesting. [Here](https://github.com/aws/aws-for-fluent-bit/pull/124/files#diff-c63759f58a022cf694046a961470c6e28c042d8f221646bc5ddbb5fa959dda75R255) they did the same but for regional ECR I guess. Do we have the error log?

Comment (zhonghui12) [reply scripts/publish.sh:131]: yes we have.
Screen Shot 2020-12-18 at 1 55 43 PM

Comment (zhonghui12) [reply scripts/publish.sh:131]: You can see here. I didn't log in again in my previous code and it succeeds with `tag: 2.10.0` but failed with `tag:latest`. The error info is `denied: Not Authorized`.

**Comment by hossain-rayhan** [comment]
I think we also need to publish the SSM parameter for the ECR public repo. We don't need any additional pipeline stage but have to update the publish_ssm script.

**Comment by PettitWesley** [comment]
@hossain-rayhan @zhonghui12  I don't think there's any value in publishing an SSM public parameter for the public repo.

The SSM parameters are needed for the private repos since they contain an opaque account ID which is different in each region. For the public repo, the link never changes, so we don't need a parameter.

File: scripts/publish.sh
Comment (PettitWesley) [inline scripts/publish.sh:529]: nit: is the argument `amazon/aws-for-fluent-bit` to the function actually needed? I think it isn't

**Review by PettitWesley — APPROVED** [review]

**Review by PettitWesley — APPROVED** [review]
Needs @hossain-rayhan review/approval

**Review by hossain-rayhan — APPROVED** [review]

**Pull request merged**
